R/plotScoreHeatmap.R

```diff
@@ -122,26 +124,39 @@ plotScoreHeatmap <- function(results, cells.use = NULL, labels.use = NULL,
     scores <- results$scores
     rownames(scores) <- rownames(results)
 
-    # Trim the scores by requested cells or labels
+    # Trim the scores and potential ordering vars by requested cells or labels
+    labels <- results$labels
     if (!is.null(cells.use)) {
         scores <- scores[cells.use,,drop=FALSE]
         clusters <- clusters[cells.use]
+        labels <- labels[cells.use]
         cells.order <- cells.order[cells.use]
     }
     if (!is.null(labels.use)) {
         scores <- scores[,labels.use,drop=FALSE]
     }
 
     # Determining how to order the cells.
-    cluster_cols <- FALSE
-    if (order.by.clusters) {
-        order <- order(clusters)
-    } else if (!is.null(cells.order)) {
-        order <- order(cells.order)
-    } else {
-        # no ordering requested
+    #   Simply use current order if clustering turned on.
+    #   Otherwise, by cells.order, if provided, else by order.by(=Labels, default).
+    if (cluster_cols) {
         order <- seq_len(nrow(scores))
-        cluster_cols <- TRUE
+    } else {
+        if (!is.null(cells.order)) {
+            order <- order(cells.order)
+        } else {
+            order.stat <- switch(match.arg(order.by),
+                labels=labels,
+                clusters=clusters
+            )
+            if (is.null(order.stat)) {
+                stop(sprintf(
+                    paste0("%s input is required when ",
+                        sQuote("order.by = %s")),
+                    sQuote("clusters"), dQuote("clusters")))
```

Comment:
TBH I would probably just hard-code it in, `sQuote` and friends only make sense if you don't know the value:

```r
stop("'clusters' input is required when 'order.by=\"clusters\"'")
```

which is somewhat more succinct.

Reply:
Just to confirm for my own knowledge moving forward, these bits in the WRE "Since ASCII does not contain directional quotation marks, it is best to use ‘'’ and let the translator (including automatic translation) use directional quotations where available." and "In R messages it is also possible to use sQuote or dQuote as in..." can be ignored because when I type ` or " it is the proper format?

Reply:
I was kind of thinking so. I mean, I guess if you want your error messages to have pretty directionality in their quotation marks, I suppose you could use `sQuote` and friends, but you can see how much more complicated it is.

Reply:
It is certainly more complicated to code!  And just for the directionality of quotation marks, I don't care.  My main concern was just whether ' and " will error in translation to other languages as we certainly have international users.  The wording of the WRE about that sounds more like a suggestion than a need, so I'll switch to the simpler style.